qadence/backends/pulser/pulses.py

```diff
@@ -118,6 +121,11 @@ def add_pulses(
 
     # TODO: lets move those to `@singledipatch`ed functions
     if isinstance(block, WaitBlock):
+        if not block.add_pattern:
+            logger.warning(
+                "Found block with `add_pattern = False`. This is not yet supported in the Pulser "
+                "backend. If an addressing pattern is specified, it will be added to all blocks."
+            )
```

Comment:
I thought the implementation would have not been so tricky given that the patterns are in a completely separate channel (the DMM). What is the main hurdle here @vytautas-a?

Reply:
If I understood the code correctly, currently the addressing pattern is a separate function that acts on the fully built sequence and adds the pattern pulses for the full duration of the sequence. To implement this there needs to be some refactoring so that the pattern is added when the actual pulse for each block is built, such that we can check if `block.add_pattern = True` and then add the pattern pulse for the duration of that block.

Reply:
I see. I will check the code a bit more in detail.